```
Error running mint automation
+ mkdir -p 16598-3c21599/gopath/src/github.com/minio
+ git -C 16598-3c21599/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 16598-3c21599/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 16598-3c21599/gopath/src/github.com/minio/minio fetch --quiet minio pull/16598/head:pr16598
+ git -C 16598-3c21599/gopath/src/github.com/minio/minio checkout --quiet pr16598
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/16598-3c21599/gopath
+ make -C 16598-3c21599/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/mux v1.9.0
go: downloading github.com/minio/pkg v1.6.1
go: downloading github.com/cespare/xxhash/v2 v2.2.0
go: downloading github.com/klauspost/compress v1.15.15
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading cloud.google.com/go/storage v1.29.0
go: downloading github.com/minio/minio-go/v7 v7.0.47
go: downloading github.com/klauspost/filepathx v1.1.1
go: downloading github.com/minio/selfupdate v0.6.0
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/minio/sio v0.3.0
go: downloading github.com/coreos/go-oidc v2.2.1+incompatible
go: downloading github.com/miekg/dns v1.1.50
go: downloading github.com/minio/xxml v0.0.3
go: downloading github.com/lithammer/shortuuid/v4 v4.0.0
go: downloading github.com/pierrec/lz4 v2.6.1+incompatible
go: downloading github.com/minio/console v0.23.1
go: downloading github.com/klauspost/readahead v1.4.0
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/minio/zipindex v0.3.0
go: downloading github.com/philhofer/fwd v1.1.2
go: downloading github.com/klauspost/reedsolomon v1.11.6
go: downloading github.com/prometheus/client_golang v1.14.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/minio/dperf v0.4.2
go: downloading github.com/minio/cli v1.24.2
go: downloading github.com/prometheus/client_model v0.3.0
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading github.com/minio/madmin-go/v2 v2.0.11
go: downloading github.com/golang-jwt/jwt/v4 v4.4.3
go: downloading github.com/rs/cors v1.8.3
go: downloading github.com/shirou/gopsutil/v3 v3.23.1
go: downloading go.etcd.io/etcd/api/v3 v3.5.7
go: downloading github.com/rs/dnscache v0.0.0-20211102005908-e0241e321417
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/djherbis/atime v1.1.0
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/fatih/color v1.14.1
go: downloading github.com/cosnicolaou/pbzip2 v1.0.1
go: downloading github.com/prometheus/procfs v0.9.0
go: downloading go.etcd.io/etcd/client/v3 v3.5.7
go: downloading github.com/tidwall/gjson v1.14.4
go: downloading github.com/inconshreveable/mousetrap v1.1.0
go: downloading github.com/Azure/azure-storage-blob-go v0.15.0
go: downloading github.com/tinylib/msgp v1.1.8
go: downloading github.com/bits-and-blooms/bloom/v3 v3.3.1
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/prometheus/common v0.39.0
go: downloading github.com/zeebo/xxh3 v1.0.2
go: downloading github.com/dchest/siphash v1.2.3
go: downloading github.com/go-openapi/loads v0.21.2
go: downloading github.com/felixge/fgprof v0.9.3
go: downloading github.com/minio/kes v0.22.3
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/dustin/go-humanize v1.0.1
go: downloading go.uber.org/atomic v1.10.0
go: downloading golang.org/x/crypto v0.5.0
go: downloading golang.org/x/oauth2 v0.4.0
go: downloading golang.org/x/sys v0.4.0
go: downloading google.golang.org/api v0.109.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/go-ldap/ldap/v3 v3.4.4
go: downloading go.uber.org/zap v1.24.0
go: downloading golang.org/x/time v0.3.0
go: downloading github.com/buger/jsonparser v1.1.1
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/coredns/coredns v1.10.0
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/minio/simdjson-go v0.4.2
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/Shopify/sarama v1.38.1
go: downloading github.com/eclipse/paho.mqtt.golang v1.4.2
go: downloading github.com/elastic/go-elasticsearch/v7 v7.17.7
go: downloading github.com/go-sql-driver/mysql v1.7.0
go: downloading github.com/lib/pq v1.10.7
go: downloading github.com/gomodule/redigo v1.8.9
go: downloading github.com/nats-io/nats.go v1.23.0
go: downloading github.com/nats-io/stan.go v0.10.4
go: downloading github.com/nsqio/go-nsq v1.1.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/rabbitmq/amqp091-go v1.6.1
go: downloading github.com/xdg/scram v1.0.5
go: downloading github.com/mattn/go-colorable v0.1.13
go: downloading github.com/mattn/go-isatty v0.0.17
go: downloading github.com/golang/protobuf v1.5.2
go: downloading golang.org/x/sync v0.1.0
go: downloading github.com/bits-and-blooms/bitset v1.5.0
go: downloading aead.dev/minisign v0.2.0
go: downloading github.com/mattn/go-runewidth v0.0.14
go: downloading github.com/tidwall/match v1.1.1
go: downloading github.com/tidwall/pretty v1.2.1
go: downloading github.com/pquerna/cachecontrol v0.1.0
go: downloading gopkg.in/square/go-jose.v2 v2.6.0
go: downloading github.com/fraugster/parquet-go v0.12.0
go: downloading github.com/minio/csvparser v1.0.0
go: downloading github.com/alecthomas/participle v0.7.1
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading golang.org/x/net v0.5.0
go: downloading github.com/google/pprof v0.0.0-20230131232505-5a9e8f65f08f
go: downloading github.com/klauspost/cpuid/v2 v2.2.3
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20221128193559-754e69321358
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.4
go: downloading go.etcd.io/etcd/client/pkg/v3 v3.5.7
go: downloading google.golang.org/grpc v1.52.3
go: downloading cloud.google.com/go v0.109.0
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/golang/snappy v0.0.4
go: downloading github.com/xdg/stringprep v1.0.3
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.4
go: downloading github.com/rivo/uniseg v0.4.3
go: downloading google.golang.org/genproto v0.0.0-20230202175211-008b39050e57
go: downloading github.com/coreos/go-semver v0.3.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/gorilla/websocket v1.5.0
go: downloading github.com/go-openapi/analysis v0.21.4
go: downloading github.com/go-openapi/spec v0.20.8
go: downloading github.com/go-openapi/swag v0.22.3
go: downloading github.com/Azure/azure-pipeline-go v0.2.3
go: downloading google.golang.org/protobuf v1.28.1
go: downloading github.com/lestrrat-go/jwx v1.2.25
go: downloading github.com/montanaflynn/stats v0.7.0
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/rjeczalik/notify v0.9.3
go: downloading github.com/beorn7/perks v1.0.1
go: downloading gopkg.in/ini.v1 v1.67.0
go: downloading github.com/minio/md5-simd v1.1.2
go: downloading github.com/rs/xid v1.4.0
go: downloading go.uber.org/multierr v1.9.0
go: downloading aead.dev/mem v0.2.0
go: downloading github.com/tklauser/go-sysconf v0.3.11
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/nats-io/nkeys v0.3.0
go: downloading golang.org/x/text v0.6.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/eapache/go-resiliency v1.3.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20230111030713-bf00bc1b83b6
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/jcmturner/gofork v1.7.6
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.3
go: downloading github.com/pierrec/lz4/v4 v4.1.17
go: downloading github.com/coreos/go-systemd/v22 v22.5.0
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/mailru/easyjson v0.7.7
go: downloading github.com/rcrowley/go-metrics v0.0.0-20201227073835-cf1acfcdf475
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/tklauser/numcpus v0.6.0
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/hashicorp/go-uuid v1.0.3
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading github.com/apache/thrift v0.17.0
go: downloading github.com/go-openapi/strfmt v0.21.3
go: downloading github.com/go-openapi/jsonpointer v0.19.6
go: downloading github.com/lestrrat-go/blackmagic v1.0.1
go: downloading github.com/lestrrat-go/iter v1.0.2
go: downloading github.com/lestrrat-go/backoff/v2 v2.0.8
go: downloading github.com/lestrrat-go/httpcc v1.0.1
go: downloading github.com/lestrrat-go/option v1.0.1
go: downloading github.com/go-openapi/jsonreference v0.20.2
go: downloading cloud.google.com/go/compute/metadata v0.2.3
go: downloading cloud.google.com/go/iam v0.10.0
go: downloading golang.org/x/xerrors v0.0.0-20220907171357-04be3eba64a2
go: downloading go.opencensus.io v0.24.0
go: downloading github.com/googleapis/gax-go/v2 v2.7.0
go: downloading github.com/asaskevich/govalidator v0.0.0-20210307081110-f21760c49a8d
go: downloading github.com/mitchellh/mapstructure v1.5.0
go: downloading github.com/go-openapi/errors v0.20.3
go: downloading go.mongodb.org/mongo-driver v1.11.1
go: downloading github.com/oklog/ulid v1.3.1
go: downloading cloud.google.com/go/compute v1.18.0
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: downloading github.com/go-openapi/runtime v0.25.0
go: downloading github.com/jessevdk/go-flags v1.5.0
go: downloading github.com/go-openapi/validate v0.22.1
go: downloading github.com/minio/mc v0.0.0-20230203133213-3aebb3362b18
go: downloading github.com/unrolled/secure v1.13.0
go: downloading github.com/minio/websocket v1.6.0
go: downloading github.com/docker/go-units v0.5.0
go: downloading github.com/jedib0t/go-pretty/v6 v6.4.4
go: downloading github.com/charmbracelet/bubbletea v0.23.1
go: downloading github.com/charmbracelet/bubbles v0.15.0
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading github.com/rivo/tview v0.0.0-20230203122838-f0550c7918da
go: downloading github.com/pkg/xattr v0.4.9
go: downloading github.com/minio/filepath v1.0.0
go: downloading github.com/posener/complete v1.2.3
go: downloading github.com/prometheus/prom2json v1.3.2
go: downloading github.com/muesli/reflow v0.3.0
go: downloading github.com/charmbracelet/lipgloss v0.6.0
go: downloading golang.org/x/term v0.4.0
go: downloading github.com/minio/colorjson v1.0.4
go: downloading github.com/gdamore/tcell/v2 v2.5.4
go: downloading github.com/navidys/tvxwidgets v0.3.0
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading gopkg.in/h2non/filetype.v1 v1.0.5
go: downloading github.com/juju/ratelimit v1.0.2
go: downloading github.com/muesli/termenv v0.13.0
go: downloading github.com/containerd/console v1.0.3
go: downloading github.com/mattn/go-localereader v0.0.1
go: downloading github.com/muesli/ansi v0.0.0-20221106050444-61f0cd9a192a
go: downloading github.com/muesli/cancelreader v0.2.2
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
go: downloading github.com/aymanbagabas/go-osc52 v1.2.1
go: downloading github.com/gdamore/encoding v1.0.0
go: downloading github.com/googleapis/enterprise-certificate-proxy v0.2.1
go: downloading github.com/google/go-cmp v0.5.9
# github.com/minio/minio/internal/s3select
internal/s3select/select.go:686:4: l.Once undefined (type *limitedReadCloser has no field or method Once, but does have once)
make: *** [build] Error 2

```

--- a/cmd/object-handlers.go
+++ b/cmd/object-handlers.go
@@ -242,7 +242,7 @@ func (api objectAPIHandlers) SelectObjectContentHandler(w http.ResponseWriter, r
 		},
 		actualSize,
 	)
-
+	defer objectRSC.Close()
 	s3Select, err := s3select.NewS3Select(r.Body)
 	if err != nil {
 		if serr, ok := err.(s3select.SelectError); ok {
@@ -3124,7 +3124,7 @@ func (api objectAPIHandlers) PostRestoreObjectHandler(w http.ResponseWriter, r *
 				},
 				actualSize,
 			)
-
+			defer objectRSC.Close()
 			if err = rreq.SelectParameters.Open(objectRSC); err != nil {
 				if serr, ok := err.(s3select.SelectError); ok {
 					encodedErrorResponse := encodeResponse(APIErrorResponse{

--- a/internal/s3select/select.go
+++ b/internal/s3select/select.go
@@ -677,18 +677,19 @@ func NewS3Select(r io.Reader) (*S3Select, error) {
 type limitedReadCloser struct {
 	io.LimitedReader
 	io.Closer
-	// for io.Closer
-	once sync.Once
+	// io.Closer can be closed idempotently multiple times
+	closerOnce sync.Once
+	// Error storing io.Closer.Close()
+	closerErr error
 }
 
 func (l *limitedReadCloser) Close() error {
-	var err error
-	l.once.Do(func() {
+	l.closerOnce.Do(func() {
 		if l.Closer != nil {
-			err = l.Closer.Close()
+			l.closerErr = l.Closer.Close()
 		}
 	})
-	return err
+	return l.closerErr
 }
 
 func newLimitedReadCloser(r io.ReadCloser, n int64) *limitedReadCloser {

--- a/internal/s3select/select.go
+++ b/internal/s3select/select.go
@@ -678,7 +678,7 @@ type limitedReadCloser struct {
 	io.LimitedReader
 	io.Closer
 	// for io.Closer
-	sync.Once
+	once sync.Once
 }
 
 func (l *limitedReadCloser) Close() error {

